.circleci/config.yml

```diff
@@ -0,0 +1,164 @@
+version: 2
+
+jobs:
+  checkout_code:
+    docker:
+      - image: jfullaondo/datadog-agent-builder-deb:0.0.2
+    working_directory: /go/src/github.com/DataDog/datadog-agent
+    steps:
+      - checkout
+      - run:
+          name: save SHA to a file
+          command: echo $CIRCLE_SHA1 > .circle-sha
+      - save_cache:
+          key: v1-repo-{{ checksum ".circle-sha" }}
+          paths:
+            - /go/src/github.com/DataDog/datadog-agent
+
+  dependencies:
+    docker:
+      - image: jfullaondo/datadog-agent-builder-deb:0.0.2
+    working_directory: /go/src/github.com/DataDog/datadog-agent
+    steps:
+      - run:
+          name: save SHA to a file
+          command: echo $CIRCLE_SHA1 > .circle-sha
+      - restore_cache:
+          keys:
+            - v1-repo-{{ checksum ".circle-sha" }}
+      - run:
+          name: grab go deps
+          command: invoke deps
+      - save_cache:
+          key: v1-govendordeps-{{ checksum "Gopkg.lock" }}
+          paths:
+            - /go/src/github.com/DataDog/datadog-agent/vendor
+      - save_cache:
+          key: v1-gobindeps
+          paths:
+            - /go/bin
+
+  unit_tests:
+    docker:
+      - image: jfullaondo/datadog-agent-builder-deb:0.0.2
+        environment:
+          USE_SYSTEM_LIBS: "1"
+    working_directory: /go/src/github.com/DataDog/datadog-agent
+    steps:
+      - run:
+          name: save SHA to a file
+          command: echo $CIRCLE_SHA1 > .circle-sha
+      - restore_cache:
+          keys:
+            - v1-repo-{{ checksum ".circle-sha" }}
+      - restore_cache:
+          keys:
+            - v1-govendordeps-{{ checksum "Gopkg.lock" }}
+      - restore_cache:
+          keys:
+            - v1-gobindeps
+      - run:
+          name: set non-free sources
+          command: sed -i 's/main/main contrib non-free/' /etc/apt/sources.list
+      - run:
+          name: install mibs for tests
+          command: apt-get update && apt-get install -q -y snmp-mibs-downloader
+      - run:
+          name: run unit tests
+          command: invoke -e test --coverage
```

Comment:
Is removing race detection a goal or an oversight?

Pierre suggested to first run the test suite without race not coverage to catch most failure cases, then re-run with both for completeness. Workflow would become:

- checkout
- deps
- fast tests
- in parallel:
  - coverage+race tests
  - integration tests
  - binary builds
- (package/images build to be added later)